platforms/jvm/plugins-application/build.gradle.kts

```diff
@@ -16,6 +16,7 @@
 
 plugins {
     id("gradlebuild.distribution.api-java")
+    id("gradlebuild.instrumented-project")
```

Comment:
❓ Shouldn't all `api-java` projects be instrumented?

Reply:
That would add annotation processor to lots of projects that don't need it. So I think it's better we add this `gradlebuild.instrumented-project` only to projects that needs it.

Reply:
So this is a performance optimization. Is it necessary? What would be the difference if we enabled annotation processing on all our API-contributing subprojects?